File: app-core/src/main/java/ru/micron/persistence/repository/UserRepository.java
package ru.micron.persistence.repository;

import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.micron.persistence.model.User;

@Repository
public interface UserRepository extends JpaRepository<User, UUID> {

  Optional<User> findUserByUsername(String username);
}

File: app-core/src/main/kotlin/ru/micron/persistence/repository/UserRepository.kt
package ru.micron.persistence.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.micron.persistence.model.User
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, UUID> {
    fun findUserByUsername(username: String): Optional<User>
    fun existsUserByUsername(username: String): Boolean
}